File: cdm/java/mil/tatrc/physiology/datamodel/properties/SEScalarMassPerAreaTime.java
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

package mil.tatrc.physiology.datamodel.properties;

import com.kitware.physiology.cdm.Properties.ScalarMassPerAreaTimeData;

import mil.tatrc.physiology.datamodel.properties.CommonUnits.MassPerAreaTimeUnit;

/**
 * @author abray
 * MassPerAreaTime Scalar Class, will enforce that units are proper to MassPerAreaTimes
 */

public class SEScalarMassPerAreaTime extends SEScalar
{
  public SEScalarMassPerAreaTime()
  {
    super();
  }
  
  /**
   * @param value
   * @param unit - specific any valid standard unit abbreviation
   * http://www.bipm.org/en/si/ for this quantity type
   */
  public SEScalarMassPerAreaTime(double value, String unit)
  {
    this();
    this.setValue(value, unit);
  }
  
  /**
   * @param value
   * @param unit - enumeration of commonly used units for this type
   * Units are not limited to this set.
   * You can use the other method convention:(double,string)
   * and specify any valid standard unit abbreviation sting for this type
   * Unit Abbreviation Standards: http://www.bipm.org/en/si/
   */
  public SEScalarMassPerAreaTime(double value, MassPerAreaTimeUnit unit)
  {
    this();
    this.setValue(value,unit);
  }
  
  public static void load(ScalarMassPerAreaTimeData src, SEScalarMassPerAreaTime dst)
  {
    SEScalar.load(src.getScalarMassPerAreaTime(),dst);
  }
  public static ScalarMassPerAreaTimeData unload(SEScalarMassPerAreaTime src)
  {
    if(!src.isValid())
      return null;
    ScalarMassPerAreaTimeData.Builder dst = ScalarMassPerAreaTimeData.newBuilder();
    unload(src,dst);
    return dst.build();
  }
  protected static void unload(SEScalarMassPerAreaTime src, ScalarMassPerAreaTimeData.Builder dst)
  {
    SEScalar.unload(src,dst.getScalarMassPerAreaTimeBuilder());
  }
  
  /**
   * @param value
   * @param unit - enumeration of commonly used units for this type
   * Units are not limited to this set.
   * You can use the other method convention:(double,string)
   * and specify any valid standard unit abbreviation sting for this type
   * Unit Abbreviation Standards: http://www.bipm.org/en/si/
   */
  public void setValue(double value, MassPerAreaTimeUnit unit)
  {
    this.setValue(value,unit.toString());
  }
  
  /**
   * @param value
   * @param unit - enumeration of commonly used units for this type
   * Units are not limited to this set.
   * You can use the other method convention:(double,string)
   * and specify any valid standard unit abbreviation sting for this type
   * Unit Abbreviation Standards: http://www.bipm.org/en/si/
   */
  public double getValue(MassPerAreaTimeUnit unit)
  {
    return this.getValue(unit.toString());
  }

  public boolean validUnit(String unit)
  {
    if(MassPerAreaTimeUnit.validUnit(unit))
      return true;
    return false;
  }
}

File: cdm/cpp/properties/SEScalarMassPerAreaTime.cpp
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

#include "stdafx.h"
#include "properties/SEScalarMassPerAreaTime.h"

const MassPerAreaTimeUnit MassPerAreaTimeUnit::g_Per_cm2_s("g/cm^2 s");

bool MassPerAreaTimeUnit::IsValidUnit(const std::string& unit)
{
  if (g_Per_cm2_s.GetString().compare(unit) == 0)
    return true;
  return false;
}

const MassPerAreaTimeUnit& MassPerAreaTimeUnit::GetCompoundUnit(const std::string& unit)
{
  if (g_Per_cm2_s.GetString().compare(unit) == 0)
    return g_Per_cm2_s;
  std::stringstream err;
  err << unit << " is not a valid MassPerAreaTime unit";
  throw CommonDataModelException(err.str());
}

void SEScalarMassPerAreaTime::Load(const cdm::ScalarMassPerAreaTimeData& src, SEScalarMassPerAreaTime& dst)
{
  SEScalarMassPerAreaTime::Serialize(src, dst);
}
void SEScalarMassPerAreaTime::Serialize(const cdm::ScalarMassPerAreaTimeData& src, SEScalarMassPerAreaTime& dst)
{
  SEScalarQuantity<MassPerAreaTimeUnit>::Serialize(src.scalarmassperareatime(), dst);
}

cdm::ScalarMassPerAreaTimeData* SEScalarMassPerAreaTime::Unload(const SEScalarMassPerAreaTime& src)
{
  if (!src.IsValid())
    return nullptr;
  cdm::ScalarMassPerAreaTimeData* dst = new cdm::ScalarMassPerAreaTimeData();
  Serialize(src, *dst);
  return dst;
}
void SEScalarMassPerAreaTime::Serialize(const SEScalarMassPerAreaTime& src, cdm::ScalarMassPerAreaTimeData& dst)
{
  SEScalarQuantity<MassPerAreaTimeUnit>::Serialize(src, *dst.mutable_scalarmassperareatime());
}